lean4-infoview/src/infoview/interactiveCode.tsx

```diff
@@ -68,18 +69,57 @@ function TypePopupContents({pos, info, redrawTooltip}: {pos: DocumentPosition, i
   } else return <>Loading..</>
 }
 
-/** Tags in code represent values which can be hovered over to display extra info. */
-function InteractiveCodeTag({pos, tag: ct, fmt}: InteractiveTagProps<CodeToken>) {
+/** Tagged spans can be hovered over to display extra info stored in the associated `SubexprInfo`. */
+function InteractiveCodeTag({pos, tag: ct, fmt}: InteractiveTagProps<SubexprInfo>) {
   const mkTooltip = React.useCallback((redrawTooltip: () => void) =>
     <div className="font-code tl pre-wrap">
       <TypePopupContents pos={pos} info={ct.info}
         redrawTooltip={redrawTooltip} />
     </div>, [pos.uri, pos.line, pos.character, ct.info])
+
+  // We mimick the VSCode ctrl-hover and ctrl-click UI for go-to-definition
+  const rs = React.useContext(RpcContext)
+  const ec = React.useContext(EditorContext);
+  const [hoverState, setHoverState] = React.useState<HoverState>('off')
+
+  const [goToLoc, setGoToLoc] = React.useState<Location | undefined>(undefined)
+  const fetchGoToLoc = async () => {
+    if (goToLoc !== undefined) return goToLoc
```

Comment:
could the cached value ever be wrong? for example, after a Refresh File Dependencies command?  Do we need to cache?

Reply:
Hm, after a refresh the whole goal state and every message would be recreated. The `InteractiveCode` component usually has a much shorter lifetime. The caching is just for UX - by starting to fetch as soon as the mouse enters the area we can often immediately resolve to target when it is clicked.

Reply:
Got it.